backends-clickhouse/src/main/scala/org/apache/spark/sql/execution/datasources/utils/MergeTreePartsPartitionsUtil.scala

```diff
@@ -27,13 +29,60 @@ import org.apache.spark.sql.connector.read.InputPartition
 import org.apache.spark.sql.execution.datasources.{HadoopFsRelation, PartitionDirectory}
 import org.apache.spark.sql.execution.datasources.v2.clickhouse.metadata.AddMergeTreeParts
 import org.apache.spark.sql.execution.datasources.v2.clickhouse.table.ClickHouseTableV2
+import org.apache.spark.sql.types.StructType
 import org.apache.spark.util.SparkResourceUtil
 import org.apache.spark.util.collection.BitSet
 
+import io.substrait.proto.{NamedStruct, Type}
+
+import scala.collection.JavaConverters._
 import scala.collection.mutable.ArrayBuffer
 
+// scalastyle:off argcount
 object MergeTreePartsPartitionsUtil extends Logging {
 
+  val DEFAULT_ORDER_BY_KEY = "tuple()"
+
+  def genOrderByAndPrimaryKeyStr(
+      orderByKeyOption: Option[Seq[String]],
+      primaryKeyOption: Option[Seq[String]]): (String, String) = {
+    val orderByKey =
+      if (orderByKeyOption.isDefined && orderByKeyOption.get.nonEmpty) {
+        orderByKeyOption.get.mkString(",")
+      } else DEFAULT_ORDER_BY_KEY
+
+    val primaryKey =
+      if (
+        !orderByKey.equals(DEFAULT_ORDER_BY_KEY) && primaryKeyOption.isDefined &&
+        primaryKeyOption.get.nonEmpty
+      ) {
+        primaryKeyOption.get.mkString(",")
+      } else ""
+
+    (orderByKey, primaryKey)
+  }
+
+  def genNamedStructJson(tableSchema: StructType): String = {
```

Comment:
irrelevant to "MergeTreePartsPartitionsUtil"

Reply:
move to `ConverterUtils.convertNamedStructJson`